File: source/downloads/code/LeetCode/Python/Best-Time-to-Buy-and-Sell-Stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        if len(prices) == 0:
            return 0

        n = len(prices)
        
        # 每一天卖出能赚到的最大钱
        dp = [0]*n
        # 当前为止股价最低值
        cur_min = prices[0]
        
        for p in range(len(prices)):
            if prices[p] <= cur_min:
                dp[p] = 0
                cur_min = prices[p]
            else:
                dp[p] = prices[p] - cur_min
        return max(dp)
        

File: downloads/code/LeetCode/Java/Best-Time-to-Buy-and-Sell-Stock.java
public class Solution {
    public int maxProfit(int[] prices) {
        int ret = 0;
        for(int i=1;i<prices.length;i++){
            if(prices[i-1] < prices[i])
                ret += prices[i] - prices[i-1];
        }   
        return ret;
    }
}
